spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -324,4 +324,8 @@ public boolean caseSensitive() {
         .defaultValue(SQLConf.CASE_SENSITIVE().defaultValueString())
         .parse();
   }
+
+  public String branch() {
+    return confParser.stringConf().option(SparkWriteOptions.BRANCH).parseOptional();
```

Comment:
I think it would be cleaner and better to handle branch explicitly in the conf rather than passing through options.

Reply:
My understanding when taking over the code was that we eventually still want to have a write options for the feature, thus this approach was used. I can change to directly pass branch into the read and write confs instead.